src/plugins/discover/public/application/main/components/no_results/no_results_suggestions/no_results_suggestion_default.tsx

```diff
@@ -8,16 +8,29 @@
 
 import React from 'react';
 import { FormattedMessage } from '@kbn/i18n-react';
-import { EuiDescriptionList, EuiDescriptionListDescription } from '@elastic/eui';
+import { EuiDescriptionList, EuiDescriptionListDescription, EuiCode } from '@elastic/eui';
+import { useInternalStateSelector } from '../../../services/discover_internal_state_container';
 
 export function NoResultsSuggestionDefault() {
+  const dataViewPattern = useInternalStateSelector((state) => state.dataView?.getIndexPattern?.());
+
   return (
     <EuiDescriptionList compressed>
       <EuiDescriptionListDescription data-test-subj="discoverNoResultsCheckIndices">
-        <FormattedMessage
-          id="discover.noResults.noDocumentsOrCheckPermissionsDescription"
-          defaultMessage="Make sure you have permission to view the indices and that they contain documents."
-        />
+        {dataViewPattern ? (
+          <FormattedMessage
+            id="discover.noResults.noDocumentsOrCheckIndicesAndPermissionsDescription"
+            defaultMessage="Make sure the indices matching {dataViewPattern} exist, that you have permission to view them and that they contain documents."
```

Comment:
My suggestion - although it should probably get 👀 from @gchaps - `Make sure data view {dataViewName} with index pattern {dataViewPattern} has matching indices and documents and that you have permission to view them.`

---

Does this work properly for text based languages?

Reply:
@mattkime Thanks, updated!

<img width="400" alt="Screenshot 2023-01-31 at 10 25 25" src="https://user-images.githubusercontent.com/1415710/215720800-21f9ff44-2c4a-463b-93e7-b5788091b4b6.png">

For SQL mode we show an error in this case rather than the message:
<img width="400" alt="Screenshot 2023-01-31 at 10 15 15" src="https://user-images.githubusercontent.com/1415710/215720920-d8125016-1cb3-48d6-8310-d63956e960ad.png">

Reply:
My suggestion is to add the 2 words in bold:

>Make sure **that the** data view {dataViewName} with index pattern {dataViewPattern} has matching indices and documents and that you have permission to view them.

Reply:
Also, can we update the error message:

Instead of 

> We encountered an error retrieving search results

how about

> Unable to retrieve search results

Reply:
Thanks @gchaps !